file/join.rs

```diff
@@ -404,13 +417,119 @@ fn process_files(
     Ok(())
 }
 
+// version 3
+fn process_files2(
+    file1_path: &PathBuf,
+    file2_path: &PathBuf,
+    sep: char,
+    field1: usize,
+    field2: usize,
+    a: u8,
+    e: Option<String>,
+    o: Option<Vec<String>>,
+    v: u8
+) -> Result<(), Box<dyn std::error::Error>> {
+    // code to support stdin
+    let stdin = io::stdin();
+    let file1: Box<dyn BufRead> = if file1_path.to_str() == Some("-") {
+        Box::new(stdin.lock())
+    } else {
+        Box::new(BufReader::new(File::open(file1_path)?))
+    };
+
+    let file2: Box<dyn BufRead> = if file2_path.to_str() == Some("-") {
+        Box::new(stdin.lock())
+    } else {
+        Box::new(BufReader::new(File::open(file2_path)?))
+    };
+
+    let mut file2_cache = vec![];
+    for line in file2.lines() {
+        let line = line?;
+        let fields2 = parse_fields(&line, sep);
+        file2_cache.push(fields2);
+    }
```

Comment:
file2_cache can lead to memory overflow and crash the app. Correct me if I'm wrong.